### PR TITLE
docs: Add matrix version between envoy and cilium

### PR DESCRIPTION
**Pull request by sayboras**
This is to add a small doc for the version matrix between Cilium and Cilium envoy versions, which is useful with the upcoming work to move the envoy proxy out of Cilium agent container.

### CODE REVIEW

**Review by jrajahalme — APPROVED** [review]

**Comment by sayboras** [comment]
This is only for documentation update, so full CI is not required.

File: Documentation/security/network/proxy/envoy.rst
Comment (zacharysarah) [inline Documentation/security/network/proxy/envoy.rst]: Be specific: avoid "this" and "that" unless it's 💯 clear. 

I know "distribution" is overloaded; feel free to replace with a specific noun that works better.
```suggestion
Cilium uses this minimal distribution as its host proxy for enforcing HTTP and other L7 policies as specified in network policies
for the cluster. Cilium proxy is distributed within the Cilium images.
```

File: Documentation/security/network/proxy/envoy.rst
Comment (zacharysarah) [inline Documentation/security/network/proxy/envoy.rst]:
```suggestion
For more information on the version compatibility matrix, see `Cilium Proxy documentation <https://github.com/cilium/proxy#version-compatibility-matrix>`_.
```

**Review by zacharysarah — APPROVED** [review]
@sayboras Nice work, just a couple of nits. ✨ I'm approving with the understanding that changes need to happen before merge.

**Comment by sayboras** [comment]
The changes are related to docs only, full CI is not required. Reviews are in, marking this ready to merge.

**Pull request merged**
